remake.yml

```diff
@@ -32,7 +32,8 @@ targets:
     command: find_oldest_sites(states, parameter)
 
   state_tasks:
-    command: do_state_tasks(oldest_active_sites, '1_fetch/src/get_site_data.R')
```

Comment:
Question about this. This means that if the file changes at all it will trigger a rerun, right? Can we depend on the actual _function_ instead of the file?

Reply:
great question. In reality, it is the _function_ changing that will trigger a true rebuild. 

You've got a main remake and then a child remake. You put the source file in the main remake, so any kind of change to that file will trigger a re-run of `do_state_tasks`. Once inside `do_state_tasks`, the source file is used as a remake "sources" input on the child remake file. This results in the functions in there being traced to the targets and the used functions being hashed to determine change. Then a second "make" call happens within `do_state_tasks`. If you changed, for example, some comments in 1_fetch/src/get_site_data.R, the second make call will result in `OK` since the hash of the functions within that file did not change, and the build will be pretty quick. If you did make real changes to a function used within the child remake, it will re-build those targets. 

Does this make sense? it is a bit complicated because it is mixing the make-like checking of "did any part of this file change, if so, re-build" (parent remake checks 1_fetch/src/get_site_data.R) with the remake checking of "did the function hashes change, if so, re-build" (child remake, checks the functions _within_ that file).

Reply:
Yep. Makes total sense. Thanks.